lib/iris/tests/test_basic_maths.py

```diff
@@ -654,72 +654,92 @@ def test_log10(self):
         self.cube2 = iris.cube.Cube(self.data2)
+        self.cube3 = iris.cube.Cube(self.data3)
+        self.cube4 = iris.cube.Cube(self.data4)
+
+        self.ops = (operator.add, operator.sub, operator.mul, operator.truediv)
+        self.iops = (operator.iadd, operator.isub,
+                     operator.imul, operator.itruediv)
 
     def test_operator(self):
         for test_op in self.ops:
             result1 = test_op(self.cube1, self.cube2)
             result2 = test_op(self.data1, self.data2)
-
-            np.testing.assert_array_equal(result1.data, result2)
+            self.assertArrayAlmostEqual(result1.data, result2)
+            result3 = test_op(self.cube3, self.cube4)
+            result4 = test_op(self.data3, self.data4)
+            self.assertArrayAlmostEqual(result3.data, result4)
 
     def test_operator_in_place(self):
         for test_op in self.iops:
             test_op(self.cube1, self.cube2)
-            if test_op == operator.itruediv:
-                # Python3 itruediv requires floats to return floats, numpy1.10:
-                # "TypeError: ufunc 'true_divide' output (typecode 'd') could
-                # not be coerced to provided output parameter (typecode 'l')
-                # according to the casting rule ''same_kind''"
-                test_op(self.data1.astype('float'), self.data2.astype('float'))
-            else:
-                test_op(self.data1, self.data2)
-
-            np.testing.assert_array_equal(self.cube1.data, self.data1)
+            test_op(self.data1, self.data2)
+            self.assertArrayAlmostEqual(self.cube1.data, self.data1)
+            if test_op != operator.itruediv:
+                # NumPy ufunc `true_divide` does not handle int data.
+                test_op(self.cube3, self.cube4)
+                expected = test_op(self.data3, self.data4)
+                self.assertArrayAlmostEqual(self.cube3.data, expected)
 
     def test_operator_scalar(self):
         for test_op in self.ops:
             result1 = test_op(self.cube1, 2)
             result2 = test_op(self.data1, 2)
-
-            np.testing.assert_array_equal(result1.data, result2)
+            self.assertArrayAlmostEqual(result1.data, result2)
+            result3 = test_op(self.cube3, 2)
+            result4 = test_op(self.data3, 2)
+            self.assertArrayAlmostEqual(result3.data, result4)
 
     def test_operator_array(self):
         for test_op in self.ops:
             result1 = test_op(self.cube1, self.data2)
             result2 = test_op(self.data1, self.data2)
-
-            np.testing.assert_array_equal(result1.data, result2)
+            self.assertArrayAlmostEqual(result1.data, result2)
+            result3 = test_op(self.cube3, self.data4)
+            result4 = test_op(self.data3, self.data4)
+            self.assertArrayAlmostEqual(result3.data, result4)
+
+    def test_cube_itruediv__int(self):
+        test_op = self.iops[-1]  # `itruediv`
```

Comment:
@dkillick This is a bit naughty ... what if the `iops` order changes in the `setUp` ... you could always be explicit and just use `operator.itruediv` ...